src/Libraries/RevitServices/Transactions/TransactionManager.cs

```diff
@@ -96,13 +98,16 @@ private TransactionManager(ITransactionStrategy strategy)
             Strategy = strategy;
             TransactionWrapper = new TransactionWrapper();
             DoAssertInIdleThread = true;
+            ReadOnlyMode = false;
         }
 
         /// <summary>
         /// Ensure that there is a valid transaction active
         /// </summary>
         public void EnsureInTransaction(Document document)
         {
+            if (ReadOnlyMode)
+                throw new Exception("Dynamo For Revit is in read-only mode");
```

Comment:
I I wonder how this exception will behave? Will Node turn yellow and display this information? For a Python node or a Custom node, how will it be displayed?

Reply:
For a typical node it will turn yellow and display this error.   The python and custom nodes do their typical behavior of suppressing the specific error message.  I will double check was the user facing UX is

Reply:
For now with this internal method we are thinking that the application is only for running the graph in headless mode where the user would not see the impact.

Reply:
Also I am going to update the message to be a bit more precise